### DISCUSSION TITLE
Freezing entire system?

### DISCUSSION THREAD

**Discussion by spacescientist** [Cimbali/pympress#281] [General]
Hello everyone,

First of all, thanks a lot for this great tool. Ever since I found out about Pympress, I've been using it in conjunction with LuaLaTeX beamer presentations, including videos and gifs on my slides. 

It's great. I love it.

There is one thing I'd like to ask here: has someone else also encountered a problem where, going through slides, their entire system suddenly freezes (except the mouse)? I am on Ubuntu 20.04, on a powerful machine that is 1 year old.

I don't want to open it as a bug since I cannot be sure that it's linked in Pympress (I cannot get a terminal to open, and cannot check what goes wrong). However, this basically only happens to me when working with Pympress — notably, when going quickly through (heavy? [*]) slides, when I am repeatedly compiling the .tex document then checking the result of various slides rather quickly.

Has someone else encountered this?

[*] Well, not that heavy: the CPU/RAM workload is close to zero when it happens, but for some slides such as those with video you can definitely feel a small delay from one to the next.

**Comment by Cimbali** [comment]
Can you try to open up a virtual console with Ctrl+Alt+F1 (or F2, F3, …) when it freezes up? If so, you should be able to check what happens from there with text-based tools:
- `top` (or its prettier alternative `htop`) to see which processes hog up resources
- `killall pympress` then back to `Ctrl+Alt+F7` (might be `Ctrl+Alt+F2` on Ubuntu?) to see if that unlocked the situation

I think it _shouldn’t_ be a memory leak as pympress manually forces garbage collection after every document reload:

https://github.com/Cimbali/pympress/blob/8702f2e75ca9475ba8c2f8effda6888febdd8584/pympress/ui.py#L810-L813

But let’s see if we can find out more.

**Reply by Cimbali** [reply]
I’m assuming it’s not only the input devices (mouse clicks / keyboard) that freeze, but actual processes (e.g. browser, anything open at that time) that freeze too? I.e. clock not updating, music (if any) stops playing, etc?

**Reply by spacescientist** [reply]
When the problem happens, the computer no longer takes any input nor keystrokes from the keyboard. 

This is why I cannot check `dmesg`, cannot switch to a virtual console, nor look at `top`/`htop` or use `kill`/`xkill`.
It becomes fully unresponsive and truly frozen. The only way out that I have found is a hard shutdown.

It's like whatever appears on the screen has been replaced by a screenshot and does not react in any way. (The only strange exception is that the mouse cursor will still move, but clicks are no longer registered).

About the clock: good point, thank you. Next time it happens, I will make sure to check whether it keeps updating itself or not.

**Reply by spacescientist** [reply]
It's also a laptop that has to connect via WiFi and use a (proprietary) VPN. 
If it had had a fixed IP, I would have tried to SSH to it, and see what's up.

**Reply by FriedrichFroebel** [reply]
I am not sure how useful the systemd journal would be in a such a case and if it might be written, might giving you some hints as well. Nevertheless, you might have to change its configuration to keep it across reboots.

**Reply by spacescientist** [reply]
Thank you both very much for your time and suggestions. 

If it keeps happening, I might have to try to do something like that indeed. If I do and find out that it's related to pympress, I'll come back and let you know. Cheers!
